BASE/src/ServerTelemetryChannel/Implementation/Transmitter.cs

```diff
@@ -169,10 +173,84 @@ internal virtual void Enqueue(Transmission transmission)
 
             if (!this.Storage.Enqueue(transmissionGetter))
             {
+                transmission.IsFlushAsyncInProgress = false;
                 TelemetryChannelEventSource.Log.TransmitterStorageSkipped(transmission.Id);
             }
         }
 
+        internal Task<bool> FlushAsync(Transmission transmission, CancellationToken cancellationToken)
+        {
+            TaskStatus taskStatus = TaskStatus.Canceled;
+            if (!cancellationToken.IsCancellationRequested)
+            {
+                transmission.IsFlushAsyncInProgress = true;
+                this.Enqueue(transmission);
+
+                try
+                {
+                    this.Storage.IncrementFlushAsyncCounter();
```

Comment:
i think this is done to prevent items being moved from Storage to Buffer/Sender, while FlushAsync is in progress... If thats the case, then this must also be done in the `MoveTransmissionsAndWaitForSender(cancellationToken)` path as well.